### PR TITLE
[ImageUtil] Support JPEG XL image encoder/decoder

### PR DESCRIPTION
**Pull request by hsgwon**
### Description of Change ###

Support JPEG XL image encoder, decoder.
Fix header offset bug.

### API Changes ###

 - TCSACR-490

### CODE REVIEW

File: src/Tizen.Multimedia.Util/ImageUtil/ImageDecoder.cs [hunk 1]
```diff
@@ -448,4 +448,26 @@ public HeifDecoder() : base(ImageFormat.Heif)
...
         internal override int HeaderOffset => 4;
     }
+
+    /// <summary>
+    /// Provides the ability to decode the JPEG (Joint Photographic Experts Group) XL encoded images.
+    /// </summary>
+    /// <since_tizen> 10 </since_tizen>
+    public class JpegXlDecoder : ImageDecoder
+    {
+        private static readonly byte[] _header = { (byte)'J', (byte)'X', (byte)'L'};
```
Comment (jiyong2) [inline src/Tizen.Multimedia.Util/ImageUtil/ImageDecoder.cs:458]: The JPEG XL file format can take two forms:
The ISO-based jpeg xl image will start with the 'jxl'.
Another jpeg xl images called 'naked' codestream start with 0xFF0A.
Please refer to below native code.

- image_util_decode.c
static unsigned char _JXL_CODESTREAM_HEADER[] = { 0xFF, 0x0A };
{ IMAGE_UTIL_JPEG_XL, _JXL_CODESTREAM_HEADER, sizeof(_JXL_CODESTREAM_HEADER), 0 },

**Review by jiyong2 — COMMENTED** [review]
Looks good to me, but  the checkheader() should approve.
Could you please check the somment.

**Review by jiyong2 — COMMENTED** [review]
Looks good to me.

File: src/Tizen.Multimedia.Util/ImageUtil/ImageEncoder.cs [hunk 2]
```diff
@@ -514,13 +514,59 @@ public class WebPEncoder : ImageEncoder
...
+        /// </summary>
+        /// <remarks><see cref="ImageEncoder.OutputFormat"/> will be the <see cref="ImageFormat.JpegXl"/>.</remarks>
+        /// <since_tizen> 10 </since_tizen>
+        public JpegXlEncoder() :
+            base(ImageFormat.JpegXl)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="JpegXlEncoder"/> class with the information for lossless or lossy compression.
```
Comment (wiertel) [inline src/Tizen.Multimedia.Util/ImageUtil/ImageEncoder.cs:543]: You could consider simplifying the description like this:
```suggestion
        /// Initializes a new instance of the <see cref="JpegXlEncoder"/> class for lossless or lossy compression.
```

File: src/Tizen.Multimedia.Util/ImageUtil/ImageEncoder.cs [hunk 3]
```diff
@@ -514,13 +514,59 @@ public class WebPEncoder : ImageEncoder
...
+        /// <value>
+        /// The property determining whether the JpegXl compression is lossless or lossy.<br/>
+        /// The default is false(lossy).</value>
+        /// <since_tizen> 10 </since_tizen>
+        public bool Lossless { get; set; } = false;
+
+        internal override void Configure(ImageEncoderHandle handle)
+        {
+            NativeEncoder.SetLossless(handle, Lossless).
+                ThrowIfFailed("Failed to configure encoder; Lossless");
```
Comment (wiertel) [inline src/Tizen.Multimedia.Util/ImageUtil/ImageEncoder.cs:569]: it's not API related, but the message would be more useful if it included the value of `Lossless`, so you might consider change like:
```suggestion
                ThrowIfFailed($"Failed to configure encoder; Lossless={Lossless}");
```

File: src/Tizen.Multimedia.Util/ImageUtil/ImageEncoder.cs (hunk 2 above)
Comment (hsgwon) [reply src/Tizen.Multimedia.Util/ImageUtil/ImageEncoder.cs]: Fixed.

File: src/Tizen.Multimedia.Util/ImageUtil/ImageEncoder.cs (hunk 3 above)
Comment (hsgwon) [reply src/Tizen.Multimedia.Util/ImageUtil/ImageEncoder.cs]: Fixed.

**Review by aferin — APPROVED** [review]

**Pull request merged**